executor/src/witgen/mod.rs

```diff
@@ -21,12 +21,19 @@ mod expression_evaluator;
 pub mod fixed_evaluator;
 mod generator;
 mod global_constraints;
+mod identity_processor;
 mod machines;
+mod query_processor;
 mod range_constraints;
+mod rows;
 pub mod symbolic_evaluator;
 mod symbolic_witness_evaluator;
 mod util;
 
+fn leak_string(s: String) -> &'static str {
+    Box::leak(s.into_boxed_str())
+}
```

Comment:
I think we should at least have a plan to get rid of this. Is this something that would go away if the generator did not own the rows and instead held a mutable reference to some `Rows` object which owns everything?

Reply:
I'll copy-paste my answer from the chat, maybe it's better here:

> So on the memory leak: I think it would be fine for this PR, I'd add a lifetime parameter to `Cell`, `Row`, etc and it ends up being lifetime `'a` of `Generator`.
> 
> The bigger problem is using the Row abstraction in `BlockMachine` then. I think it makes sense that the machine owns its rows, but then `BlockMachine` needs a lifetime parameter and as far as I understand it needs to be `'static` for `BlockMachine` to be able to live in a box. So, I think this would need to change, and I'm not sure of the consequences.
> 
> Some other ideas:
> 
> - We could use `Rc<String>` for the names.
> - I feel like it would be fair to copy everything (including names) once per machine instance. That's what happens for everything `BlockMachine` owns, so we could copy the names as well. But struct's can't have self-referential references, so this doesn't solve the problem. I wonder if here is a way to work around that. Basically, what I want to express is: The names live in some fixed memory location for as long as `BlockMachine` lives, so `BlockMachine` can have references to this data.

Reply:
> Is this something that would go away if the generator did not own the rows and instead held a mutable reference to some Rows object which owns everything?

I think long term, it makes sense for each machine to own "its" columns, right? So every machine would own all rows, but with a row being a sparse set of columns.